Initial wrapping of the text-field

YuriyVaadin opened the pull request:
Fixes #3

tomivirkki commented:
Review status: 0 of 8 files reviewed at latest revision, all discussions resolved.

---

*[demo/time-picker-basic-demos.html, line 10 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_FQqe5qPH8f0prrSa:-LB_FQqf1mLWTjqhfTzQ:bn0x5g1) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/demo/time-picker-basic-demos.html#L10)):*
> ```HTML
> 
> 
>     <h3>Plain time picker</h3>
> ```

Demo titles should use Title Case

---

*[src/vaadin-time-picker.html, line 24 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_KS4F6LtjqIXcbZ2v:-LB_KS4F6LtjqIXcbZ2w:b2kjw1e) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L24)):*
> ```HTML
>     </style>
>     <vaadin-time-picker-text-field
>       value="[[value]]"
> ```

Because of this binding, I can programmatically set timePicker.value = "foo" and it gets reflected in the text field. Also changing the text to "bar" will revert the value back to (invalid) "foo". Add tests for this (should not allow setting invalid value programmatically).

---

*[src/vaadin-time-picker.html, line 26 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_K0Rl284sGGV9lVO-:-LB_K0Rl284sGGV9lVO0:b-elv5jm) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L26)):*
> ```HTML
>       value="[[value]]"
>     >
>       <slot name="prefix" slot="prefix"></slot>
> ```

Why is this text-field feature specifically exposed?

---

*[src/vaadin-time-picker.html, line 79 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_Izb3AFCvv1OAkXR8:-LB_Izb3AFCvv1OAkXR9:b-1luf27) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L79)):*
> ```HTML
>         }
> 
>         _setValue(e) {
> ```

If this is supposed to be an event listener function for "change" event, rather name it to "_onChange". It would be hard to guess what _setValue(event) api means.

---

*[src/vaadin-time-picker.html, line 87 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_H9dx7IBSkgYca7CH:-LB_H9dy84a-SG0NesxD:b-xcigc4) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L87)):*
> ```HTML
>           const parts = /^(\d{2}):(\d{2})$/.exec(str);
>           if (!parts) {
>             return this.value;
> ```

Doesn't make sense that a function called parseTime(str) returns this.value

---

*[src/vaadin-time-picker-text-field.html, line 3 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_FlH-6bAAdz6iDorR:-LB_FlH-6bAAdz6iDorS:bb6ht2i) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker-text-field.html#L3)):*
> ```HTML
> <!--
> @license
> Copyright (c) 2017 Vaadin Ltd.
> ```

2018

---

*[src/vaadin-time-picker-text-field.html, line 17 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_FtJ89UdDWwqHhEhD:-LB_FtJ89UdDWwqHhEhE:b-ft9foi) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker-text-field.html#L17)):*
> ```HTML
>       *
>       * See [`<vaadin-text-field>` documentation](https://github.com/vaadin/vaadin-text-field/blob/master/src/vaadin-text-field.html)
>       * for `<vaadin-dropdown-menu-text-field>` parts and available slots (prefix, suffix etc.)
> ```

vaadin-dropdown-menu-text-field?

---

*[test/time-picker-test.html, line 21 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_Gv1a8oSWKilSNwuL:-LB_Gv1a8oSWKilSNwuM:b-qswuhz) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/test/time-picker-test.html#L21)):*
> ```HTML
>   <script>
>     describe('input test', function() {
>       var timePicker, timePickerTextField;
> ```

Don't use var

---

*[test/time-picker-test.html, line 33 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_GDYDDCW_382rrZms:-LB_GDYDDCW_382rrZmt:b-4hlda2) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/test/time-picker-test.html#L33)):*
> ```HTML
> 
>       it('value property should be undefined by default', function() {
>         expect(timePicker.value).to.be.equal(undefined);
> ```

Why undefined by default. Date picker has '' as the default value: https://github.com/vaadin/vaadin-date-picker/blob/master/src/vaadin-date-picker-mixin.html#L45 Makes sense to align these

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->

web-padawan commented:
Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[src/vaadin-time-picker.html, line 85 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_QUup6PQB0U65y0d-:-LB_QUup6PQB0U65y0d0:b-ivro2s) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L85)):*
> ```HTML
>         _parseTime(str) {
>           // Parsing with RegExp to ensure correct format
>           const parts = /^(\d{2}):(\d{2})$/.exec(str);
> ```

consider not hardcoding the colon character, but storing it in the `separator` variable instead (can be only used inside of this function for now), so that it would be easier to expose it in future

---

*[test/time-picker-test.html, line 25 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_ACQcAu_6Iq9Qo0IY:-LB_ACQcAu_6Iq9Qo0IZ:b3b4c4q) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/test/time-picker-test.html#L25)):*
> ```HTML
>       beforeEach(function() {
>         timePicker = fixture('default');
>         timePickerTextField = timePicker.shadowRoot.querySelector('vaadin-time-picker-text-field');
> ```

you could just use `timePicker._inputElement`

---

*[theme/lumo/vaadin-time-picker.html, line 1 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_RLrX6pMMilA2XWfi:-LB_RLrX6pMMilA2XWfj:b46sqgq) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/theme/lumo/vaadin-time-picker.html#L1)):*
> ```HTML
> <link rel="import" href="../../../vaadin-text-field/theme/lumo/vaadin-text-field.html">
> ```

should we create ``

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->

web-padawan commented:
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[theme/lumo/vaadin-time-picker.html, line 1 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_RLrX6pMMilA2XWfi:-LB_S0joBcqIOMKl4xPt:b-utd0av) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/theme/lumo/vaadin-time-picker.html#L1)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

should we create ``
</blockquote></details>

... create "vaadin-time-picker-styles.html" and place this import there, as we do for other components in the p3-preview branches?

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
Review status: 4 of 8 files reviewed at latest revision, 12 unresolved discussions.

---

*[demo/time-picker-basic-demos.html, line 10 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_FQqe5qPH8f0prrSa:-LB_Pb88E6VTVZhnzp81:b-bw8e14) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/demo/time-picker-basic-demos.html#L10)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Demo titles should use Title Case
</blockquote></details>

Updated.

---

*[src/vaadin-time-picker.html, line 24 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_KS4F6LtjqIXcbZ2v:-LB_UPb453WfrpQLkfVJ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L24)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Because of this binding, I can programmatically set timePicker.value = "foo" and it gets reflected in the text field. Also changing the text to "bar" will revert the value back to (invalid) "foo". Add tests for this (should not allow setting invalid value programmatically).
</blockquote></details>

Done.

---

*[src/vaadin-time-picker.html, line 26 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_K0Rl284sGGV9lVO-:-LB_Pa7_CHoZIgYAPydt:bss4a2b) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L26)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Why is this text-field feature specifically exposed?
</blockquote></details>

Removed.

---

*[src/vaadin-time-picker.html, line 79 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_Izb3AFCvv1OAkXR8:-LB_Pk-81GXPFOgbJsPj:b-a2ujfl) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L79)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

If this is supposed to be an event listener function for "change" event, rather name it to "_onChange". It would be hard to guess what _setValue(event) api means.
</blockquote></details>

Changed.

---

*[src/vaadin-time-picker.html, line 85 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_QUup6PQB0U65y0d-:-LB_UO04B7ZjBYiacdRJ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L85)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

consider not hardcoding the colon character, but storing it in the `separator` variable instead (can be only used inside of this function for now), so that it would be easier to expose it in future
</blockquote></details>

Done.

---

*[src/vaadin-time-picker.html, line 87 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_H9dx7IBSkgYca7CH:-LB_UKZo5f7bPy3HK_a9:bymrjec) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L87)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Doesn't make sense that a function called parseTime(str) returns this.value
</blockquote></details>

Not returning `this.value` anymore.  Working now with `_previousValue` instead.

---

*[src/vaadin-time-picker-text-field.html, line 3 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_FlH-6bAAdz6iDorR:-LB_Pow18_xdrMjF1Mdc:b-bw8e14) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker-text-field.html#L3)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

2018
</blockquote></details>

Updated.

---

*[src/vaadin-time-picker-text-field.html, line 17 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_FtJ89UdDWwqHhEhD:-LB_PqLzA-Cq9pRs7cng:b-bw8e14) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker-text-field.html#L17)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

vaadin-dropdown-menu-text-field?
</blockquote></details>

Updated.

---

*[test/time-picker-test.html, line 21 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_Gv1a8oSWKilSNwuL:-LB_PuGk1KzgqaGbFSIU:b-bw8e14) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/test/time-picker-test.html#L21)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Don't use var
</blockquote></details>

Updated.

---

*[test/time-picker-test.html, line 25 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_ACQcAu_6Iq9Qo0IY:-LB_Upj2-L2rfInthQ8d:b-896fix) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/test/time-picker-test.html#L25)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

you could just use `timePicker._inputElement`
</blockquote></details>

Done.

---

*[test/time-picker-test.html, line 33 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_GDYDDCW_382rrZms:-LB_QA6zATdnxqvfQfgP:b-bw8e14) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/test/time-picker-test.html#L33)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Why undefined by default. Date picker has '' as the default value: https://github.com/vaadin/vaadin-date-picker/blob/master/src/vaadin-date-picker-mixin.html#L45 Makes sense to align these
</blockquote></details>

Updated.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->

web-padawan commented:
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->

web-padawan commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[theme/lumo/vaadin-time-picker.html, line 1 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_RLrX6pMMilA2XWfi:-LB_bExsB87ARRNwDh9I:b-896fix) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/theme/lumo/vaadin-time-picker.html#L1)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

... create "vaadin-time-picker-styles.html" and place this import there, as we do for other components in the p3-preview branches?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[src/vaadin-time-picker.html, line 24 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_KS4F6LtjqIXcbZ2v:-LB_diX_2XdKW_ku3K9-:bk4ur0f) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L24)):*
<details><summary><i>Previously, YuriyVaadin (Yuriy Yevstihnyeyev) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Since the value property notifies, you now get "value-changed" events on every keystroke, even when the value is invalid. Drop the two-way binding and add tests that verify that value changed gets only fired when appropriate.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/vaadin-time-picker.html, line 24 at r1](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_KS4F6LtjqIXcbZ2v:-LB_ynUJ9DkCCNc9M2by:bijku87) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/45ee423634585e39d10d2ea8791f56f879547bbf/src/vaadin-time-picker.html#L24)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Since the value property notifies, you now get "value-changed" events on every keystroke, even when the value is invalid. Drop the two-way binding and add tests that verify that value changed gets only fired when appropriate.
</blockquote></details>

Dropped and tests are added.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[src/vaadin-time-picker.html, line 85 at r2](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_erWl9tgFqHjuvoMK:-LB_erWl9tgFqHjuvoML:b-tij01e) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/0aed6188fd12db91fcd4db1f9d4a0ac0d761fb51/src/vaadin-time-picker.html#L85)):*
> ```HTML
> 
>         _valueChanged(value, oldValue) {
>           if (this._inputElement.hasAttribute('focused')) {
> ```

Why is this check still needed now that 2-way binding is gone?

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/vaadin-time-picker.html, line 85 at r2](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LB_erWl9tgFqHjuvoMK:-LBdxb-14g2Cm_N7s9j8:bss4a2b) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/0aed6188fd12db91fcd4db1f9d4a0ac0d761fb51/src/vaadin-time-picker.html#L85)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Why is this check still needed now that 2-way binding is gone?
</blockquote></details>

Removed.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 8 files at r1, 2 of 4 files at r2, 2 of 2 files at r3, 1 of 2 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-:-LBdy5h_D0hx7MiISJka:bnfp4nl)*
<!-- Sent from Reviewable.io -->

web-padawan commented:
Reviewed 1 of 2 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->

platosha commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 8 files at r1, 2 of 4 files at r2, 2 of 2 files at r3, 1 of 2 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---

*[test/time-picker-test.html, line 20 at r5](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LBeKshQ17VuAl02HkYS:-LBeKshQ17VuAl02HkYT:b6tnsse) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/31c9ce8ab5585d58d3a0802796f97f90c907d516/test/time-picker-test.html#L20)):*
> ```HTML
> 
>   <script>
>     describe('input test', function() {
> ```

BTW I would <3 to have a separate test suite for every topic. Otherwise, if we stick to a single test file from the start, that quickly becomes too big to comfortably edit.

Now that the input tests are nicely grouped right away, it makes sense to extract them into something like `test/input.html`.

Not a blocker in any way, no changes required right away. This can be done later at any point.

---

*[test/time-picker-test.html, line 48 at r5](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LBeLhP8-Df_DHyeHSRI:-LBeLhP8-Df_DHyeHSRJ:b-exiupj) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/31c9ce8ab5585d58d3a0802796f97f90c907d516/test/time-picker-test.html#L48)):*
> ```HTML
>       it('should set values of inputElement and time-picker if the format is valid', function() {
>         timePicker._inputElement.value = '12:00';
>         timePicker._inputElement.dispatchEvent(new CustomEvent('change'));
> ```

BTW might make sense extract a helper method, since these lines are always in pair

```js
enterValue(value) {
  timePicker._inputElement.value = value;
  timePicker._dispatchEvent(new CustomEvent('change'));
}
```

Not a blocker though.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-:-LBeMC9Z0D8BuRfeQ39d:bnfp4nl)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[test/time-picker-test.html, line 20 at r5](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LBeKshQ17VuAl02HkYS:-LBeR02FEYez-dhv4sxz:bc8zu3v) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/31c9ce8ab5585d58d3a0802796f97f90c907d516/test/time-picker-test.html#L20)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

BTW I would <3 to have a separate test suite for every topic. Otherwise, if we stick to a single test file from the start, that quickly becomes too big to comfortably edit.

Now that the input tests are nicely grouped right away, it makes sense to extract them into something like `test/input.html`.

Not a blocker in any way, no changes required right away. This can be done later at any point.
</blockquote></details>

Will be done in scope of next PR.

---

*[test/time-picker-test.html, line 48 at r5](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13#-LBeLhP8-Df_DHyeHSRI:-LBeR2XS3nE_3GrmKsjt:bc8zu3v) ([raw file](https://github.com/vaadin/vaadin-time-picker/blob/31c9ce8ab5585d58d3a0802796f97f90c907d516/test/time-picker-test.html#L48)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

BTW might make sense extract a helper method, since these lines are always in pair

```js
enterValue(value) {
  timePicker._inputElement.value = value;
  timePicker._dispatchEvent(new CustomEvent('change'));
}
```

Not a blocker though.
</blockquote></details>

Will be done in scope of next PR.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/13)*
<!-- Sent from Reviewable.io -->
